crypto/mlkem/mlkem1024.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Copyright (c) 2024, Google Inc. */
+
+#ifndef OPENSSL_NO_MLKEM
+
+# include <crypto/mlkem.h>
+
+/* Shortened for internal consumption by BoringSSL code */
+# define DEGREE             MLKEM_DEGREE
+# define RANK               MLKEM1024_RANK
+# define ETA1               MLKEM1024_ETA1
+# define ETA2               MLKEM1024_ETA2
+# define DU                 MLKEM1024_DU
+# define DV                 MLKEM1024_DV
+# define SECBITS            MLKEM1024_RNGSEC
+# define CIPHERTEXT_BYTES   MLKEM1024_CIPHERTEXT_BYTES
+
+/* Primary key gen/encap/decap API */
+# define ossl_mlkem_generate_key                ossl_mlkem1024_generate_key
+# define ossl_mlkem_encap                       ossl_mlkem1024_encap
+# define ossl_mlkem_decap                       ossl_mlkem1024_decap
+# define ossl_mlkem_public_from_private         ossl_mlkem1024_public_from_private
+
+/* For testing and encoding rather than KEM computation */
+# define ossl_mlkem_private_key_from_seed           ossl_mlkem1024_private_key_from_seed
+# define ossl_mlkem_generate_key_external_seed      ossl_mlkem1024_generate_key_external_seed
+# define ossl_mlkem_encap_external_entropy          ossl_mlkem1024_encap_external_entropy
+# define ossl_mlkem_marshal_public_key              ossl_mlkem1024_marshal_public_key
+# define ossl_mlkem_parse_public_key                ossl_mlkem1024_parse_public_key
+# define ossl_mlkem_marshal_private_key             ossl_mlkem1024_marshal_private_key
+# define ossl_mlkem_parse_private_key               ossl_mlkem1024_parse_private_key
+
+typedef ossl_mlkem_scalar           scalar;
+typedef ossl_mlkem1024_vector       vector;
+typedef ossl_mlkem1024_matrix       matrix;
+typedef ossl_mlkem1024_public_key   public_key;
+typedef ossl_mlkem1024_private_key  private_key;
+
+# include "mlkem-variant.c"
```

Comment:
Is `.c` ideal here?
We use `.inc` more widely.

As usual, we're not consistent.